bot/cogs/weather_cog.py

```diff
@@ -0,0 +1,353 @@
+                    else:
+                        time_str = time_num.strftime("%I:%M %p (%A)")
+
+                # Temperatures
+                if req_type == 'day':
+                    day_temp = val.get('temp',{}).get('day',{})
+                    night_temp = val.get('temp',{}).get('night',{})
+                else:
+                    temp = val.get('temp',{})
+
+                # Chance of Precipitation
+                precp = val.get('pop',{})
+                
+                # Weather Conditions
+                desc = val.get('weather',{})[0].get('description',{}).title()
+
+                # Humidity
+                hum = val.get('humidity',{})
+
+                # Wind Speed and Direction
+                wind = val.get('wind_speed',{})
+                wind_deg = val.get('wind_deg',{})
+                ix = round(float(wind_deg) / (360. / len(dirs)))
+                wind_dir = dirs[ix % len(dirs)]
+
+                # Building the Page
+                if req_type == 'day':
+                    if i > 1:
+                        page += f'{date_str}: {round(day_temp,1)}°F\n'
+                        page += f'{date_str} Night: {round(night_temp,1)}°F\n'
+                    elif i == 1:
+                        page += f'Tomorrow: {round(day_temp,1)}°F\n'
+                        page += f'Tomorrow Night: {round(night_temp,1)}°F\n'
+                    else:
+                        page += f'Today: {round(day_temp,1)}°F\n'
+                        page += f'Tonight: {round(night_temp,1)}°F\n'
+                else:
+                    page += f'Time: {time_str}\n'
+                    page += f'Temperature: {round(temp,1)}°F\n'
+                
+                page += f'Condition: {desc}\n\n'
+                page += f'Chance of Precipitation: {round(precp*100)}%\n'
+                page += f'Humidity: {round(hum)}%\n'
+                page += f'Wind: {round(wind,1)} mph ({wind_dir})\n\n'
+                pages.append(page)
+        
+        return pages, num_hr, num_day
+
+    async def weatherCode(self, ctx, loc, is_cond, is_hr, is_day):
+        # Remove any characters not in ranges a-z, A-Z, or 0-9
+        # Exceptions: & _ - , and <space>
+        # per the ASCII Table https://www.asciitable.com
+        loc = re.sub("[^a-zA-Z0-9 ,&_-]+", "", loc)
+
+        # Geocoding URL
+        url_GeoAPI = f'{url_Geo}{loc}'
+
+        self.geocode_api_key = BotSecrets.get_instance().geocode_key
+        self.weather_api_key = BotSecrets.get_instance().weather_key
+
+        geo_queryparams = {
+            'auth' : self.geocode_api_key,
+            'json' : '1',
+            }
+
+        # Message to Display while APIs are called
+        wait_msg = await ctx.send('Converting location')
+
+        # Try Except for catching errors that could give away either API key
+        try:
+            async with aiohttp.request("GET", url_GeoAPI, params=geo_queryparams) as response:
```

Comment:
this flow should be broken up into different segments, instead of checking if the response == 200 you should check if it doesnt equal and send the error then, this will allow us to keep the indentation very flat

Reply:
Done.